code/ui/blocks/src/blocks/Primary.tsx

```diff
@@ -11,17 +12,58 @@ interface PrimaryProps {
    * @deprecated Primary block should only be used to render the primary story, which is automatically found.
    */
   name?: string;
+  /**
+   * Specify where to get the primary story from.
+   */
+  of?: Of;
 }
 
-export const Primary: FC<PrimaryProps> = ({ name }) => {
+const getPrimaryFromResolvedOf = (resolvedOf: ReturnType<typeof useOf>) => {
+  switch (resolvedOf.type) {
+    case 'meta': {
+      return resolvedOf.csfFile.stories[0] || null;
+    }
+    case 'component': {
+      throw new Error(
+        `Unsupported module type. Primary's \`of\` prop only supports \`meta\`, got: ${
+          (resolvedOf as any).type
+        }`
+      );
+    }
+    default: {
+      throw new Error(
+        `Unrecognized module type resolved from 'useOf', got: ${(resolvedOf as any).type}`
+      );
+    }
+  }
+};
+
+export const Primary: FC<PrimaryProps> = (props) => {
+  const { name, of } = props;
+
+  if ('of' in props && of === undefined) {
+    throw new Error('Unexpected `of={undefined}`, did you mistype a CSF file reference?');
+  }
+
   const docsContext = useContext(DocsContext);
+
+  let story;
+  if (of) {
+    const resolvedOf = useOf(of || 'meta', ['meta']);
+    story = getPrimaryFromResolvedOf(resolvedOf);
```

Comment:
I'm sorry for not realising this sooner, I just thought of it now. I think now that we supply `'meta'` as the only valid type, we don't need the "complex" error handling in `getPrimaryFromResolvedOf`, as that is already done within `useOf` based on the second argument - it should even be typed correctly.

So maybe you could try this and remove `getPrimaryFromResolvedOf` entirely?

```suggestion
    const resolvedOf = useOf(of || 'meta', ['meta']);
    story = resolvedOf.csfFile.stories[0] || null;
```

Reply:
Hey, sorry for the late reply, I just tested it and it seems to work fine. Good catch, I'll push a commit with the changes.